src/test/java/com/google/neighborgood/IntegrationTest.java

```diff
@@ -0,0 +1,1065 @@
+    assertEquals(
+        "Homepage login message as guest user", "Login to help out a neighbor!", loginElement);
+
+    By addTaskButton = By.id("addtaskbutton");
+    driver.manage().timeouts().implicitlyWait(5, TimeUnit.SECONDS);
+    List<WebElement> addTaskButtonElement = driver.findElements(addTaskButton);
+    // Add task button should be missing when user is not logged in
+    assertTrue(
+        "Add task button must not be present for guest users", addTaskButtonElement.isEmpty());
+
+    By dashboardIcon = By.className("dashboard-icon");
+    driver.manage().timeouts().implicitlyWait(5, TimeUnit.SECONDS);
+    List<WebElement> dashboardIconsElement = driver.findElements(dashboardIcon);
+    // Dashboard icons should be missing when user is not logged in
+    assertTrue(
+        "Dashboard icons must not be present for guest users", dashboardIconsElement.isEmpty());
+
+    // Element holding message displayed when there are no tasks
+    boolean taskResultsMessageDisplayed =
+        wait.until(
+                new Function<WebDriver, WebElement>() {
+                  public WebElement apply(WebDriver driver) {
+                    return driver.findElement(By.id("no-tasks-message"));
+                  }
+                })
+            .isDisplayed();
+    // Message alerting user there are no tasks nearby should be displayed since there are no tasks
+    // yet in the site
+    assertTrue("No tasks in neighborhood message should be displayed", taskResultsMessageDisplayed);
+
+    loginNewUser(USER_EMAIL, USER_NICKNAME, USER_ADDRESS, USER_PHONE, USER_ZIPCODE, USER_COUNTRY);
+    ifLaggingThenRefresh();
+
+    // After new user fills out user info, they should be redirected to userpage
+    assertTrue("User in user profile page", driver.getCurrentUrl().contains("/user_profile.jsp"));
+    assertEquals("My Account", driver.getTitle());
+
+    // Verifies logged user in userpage and then in homepage
+    verifyLoggedUser(USER_NICKNAME, "log-out-link");
+    backToHome();
+    verifyLoggedUser(USER_NICKNAME, "login-logout");
+  }
+
+  @Test
+  /** Tests functionality of adding tasks from the homepage */
+  public void _02_Homepage_AsLoggedUser_AddTask() {
+    // Confirm that the user is still logged in
+    verifyLoggedUser(USER_NICKNAME, "login-logout");
+
+    // Randomizes task contents
+    Random random = new Random();
+    String taskDetail = TASK_DETAIL + random.nextInt(1000);
+    String taskOverview = TASK_OVERVIEW + random.nextInt(1000);
+    String rewardPoints = Integer.toString(random.nextInt(201));
+    int categoryOptionIndex = random.nextInt(TASK_CATEGORIES.length);
+    String taskCategory = TASK_CATEGORIES[categoryOptionIndex];
+
+    addTask(taskDetail, rewardPoints, categoryOptionIndex, taskOverview);
+
+    // User should be redirected to user profile page after adding a task
+    assertTrue(
+        "User should be in user profile page",
+        driver.getCurrentUrl().contains("/user_profile.jsp"));
+
```

Comment:
I'm not familiar with the tools you used here, but I wonder if there is a way to check how many `<tr>` elements are present on the new user page. We can count the number of `<tr>` elements and make sure that the number of `<tr>` elements is equal to the number of tasks that we have added.

Reply:
It sounds like there should be a way to do this, I'll look into this and add this too - thanks for the suggestion!